### PR TITLE
test(notification-service): added unit testcases

### LINKED ISSUE #242: Add Unit Tests for Notification Service
Describe the feature
Add Unit Tests for Notification Service

Expected behavior
We should have at least 75% unit test case coverage in all micro services

### PR DESCRIPTION
**Pull request by ashutosh-bansal-2136**
gh-242

## Description

Unit test cases added.

Fixes #242 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [x] npm test passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/notification-service/src/__tests__/unit/channel-manager.service.unit.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  describe('returns true for channel access allowed', () => {
+    it('returns true', async () => {
+      const result = channelManagerProvider.value().isChannelAccessAllowed();
+      expect(result).to.eql(true);
+    });
+  });
+
+  function setUp() {
+    channelManagerProvider = new ChannelManagerProvider();
```
Comment (samarpan-b) [inline services/notification-service/src/__tests__/unit/channel-manager.service.unit.ts:19]: can we add some default logic of channel manager in our test case

File: services/notification-service/src/__tests__/unit/notification-user.service.unit.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+      const result = await notificationUserProvider
+        .value()
+        .getNotifUsers(notification);
+      expect(result).to.eql(notificationUser);
+    });
+  });
+
+  function setUp() {
+    notificationUserProvider = new NotificationUserProvider();
```
Comment (samarpan-b) [inline services/notification-service/src/__tests__/unit/notification-user.service.unit.ts:42]: again do we have any default impl for this ?

**Review by samarpan-b — CHANGES_REQUESTED** [review]

File: services/notification-service/package.json [hunk 3]
```diff
@@ -57,18 +58,22 @@
     "@loopback/core": "^2.16.0",
     "@loopback/repository": "^3.6.0",
     "@loopback/rest": "^9.3.0",
+    "@loopback/rest-explorer": "^3.3.2",
     "@sourceloop/core": "^2.0.1",
+    "@types/jsonwebtoken": "^8.5.4",
```
Comment (samarpan-b) [inline services/notification-service/package.json:63]: types should be moved to dev deps.

**Review by samarpan-b — CHANGES_REQUESTED** [review]

File: services/notification-service/src/__tests__/unit/channel-manager.service.unit.ts (hunk 1 above)
Comment (ashutosh-bansal-2136) [reply services/notification-service/src/__tests__/unit/channel-manager.service.unit.ts:19]: added acceptance test cases for adding mock provider

File: services/notification-service/src/__tests__/unit/notification-user.service.unit.ts (hunk 2 above)
Comment (ashutosh-bansal-2136) [reply services/notification-service/src/__tests__/unit/notification-user.service.unit.ts:42]: added acceptance test cases for adding mock provider

File: packages/core/locales/en.json [hunk 4]
```diff
@@ -16,5 +16,11 @@
 	"ENOENT": " no such file or directory, open '/home/samarpan/projects/loopback/loopback4-microservice-catalog/services/authentication-service/public/index.html'",
 	"The key 'sf.clientAuthentication.currentClient' is not bound to any value in context RequestContext-yWsyEvu3TeGuly75aFbRdg-13 (context": " RequestContext-yWsyEvu3TeGuly75aFbRdg-13, binding: sf.clientAuthentication.currentClient, resolutionPath: controllers.LoginController --> @LoginController.constructor[0])",
 	"Client Invalid": "Client Invalid",
-	"host is required": "host is required"
+	"host is required": "host is required",
```
Comment (akshatdubeysf) [inline packages/core/locales/en.json:19]: these changes are not needed.

**Review by akshatdubeysf — APPROVED** [review]

File: packages/core/locales/en.json (hunk 4 above)
Comment (ashutosh-bansal-2136) [reply packages/core/locales/en.json]: done

**Pull request merged**